test/testcafe/spec/ChallengeAuthenticatorRsa_spec.js

```diff
@@ -0,0 +1,92 @@
+import { RequestMock } from 'testcafe';
+import xhrAuthenticatorRequiredRsa from '../../../playground/mocks/data/idp/idx/authenticator-verification-rsa';
+import xhrInvalidPasscode from '../../../playground/mocks/data/idp/idx/error-authenticator-verification-rsa';
+import xhrPasscodeChange from '../../../playground/mocks/data/idp/idx/error-authenticator-verification-passcode-change-rsa';
+import xhrSuccess from '../../../playground/mocks/data/idp/idx/success';
+import ChallengeRsaPageObject from '../framework/page-objects/ChallengeOnPremPageObject';
+import SuccessPageObject from '../framework/page-objects/SuccessPageObject';
+
+const mockChallengeAuthenticatorRsa = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrAuthenticatorRequiredRsa)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(xhrSuccess);
+
+const mockInvalidPasscode = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrAuthenticatorRequiredRsa)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(xhrInvalidPasscode, 403);
+
+const mockPasscodeChange = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrAuthenticatorRequiredRsa)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(xhrPasscodeChange, 403);
+
+fixture('Challenge Authenticator RSA');
+
+async function setup(t) {
+  const challengeRsaPage = new ChallengeRsaPageObject(t);
+  await challengeRsaPage.navigateToPage();
+  return challengeRsaPage;
+}
+
+test.requestHooks(mockChallengeAuthenticatorRsa)('challenge RSA authenticator', async t => {
+  const challengeRsaPage = await setup(t);
+
+  const { log } = await t.getBrowserConsoleMessages();
+  await t.expect(log.length).eql(3);
+  await t.expect(log[0]).eql('===== playground widget ready event received =====');
+  await t.expect(log[1]).eql('===== playground widget afterRender event received =====');
+  await t.expect(JSON.parse(log[2])).eql({
+    controller: 'mfa-verify-totp',
+    formName: 'challenge-authenticator',
+    authenticatorKey: 'okta_rsa',
+    methodType: 'otp'
+  });
+
+  const pageTitle = challengeRsaPage.getPageTitle();
+  const saveBtnText = challengeRsaPage.getSaveButtonLabel();
+  await t.expect(saveBtnText).contains('Verify');
+  await t.expect(pageTitle).contains('Verify with RSA SecurID');
+
+  // verify passcode
+  await challengeRsaPage.verifyFactor('credentials.passcode', 'test');
```

Comment:
this function name is terrible... I'll file a jira to rename to `typeInputValue` or `setInputValue`, since it really doesn't verify the factor, it just fills in one input field. I was really confused on `verifyFactor` followed by a `clickNextButton`...

Reply:
Haha true, most of our tests use verifyFactor.. old LEA code i guess. Thanks @mauriciocastillosilva-okta. Let me know once you file that JIRA. I can pick it up if / when free

Reply:
https://oktainc.atlassian.net/browse/OKTA-369635